test/nonuniform.cl

```diff
@@ -0,0 +1,23 @@
+// RUN: clspv %s -o %t.spv --spv-version=1.5 --decorate-nonuniform
+// RUN: spirv-dis -o %t2.spvasm %t.spv
+// RUN: FileCheck %s < %t2.spvasm
+// RUN: spirv-val --target-env vulkan1.2spv1.5 %t.spv
+
+
+__kernel void test(__global char2 *srcA, __global char2 *srcB, __global char *srcC, __global char2 *dst)
```

Comment:
This seems to fail a couple debug builds. Is the IR consumed by the producer the same in those cases?

I'd suggest turning this in a couple .ll tests that just run the SPIR-V producer.